plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/rest/TestTrinoRESTCatalogConnectorTest.java

```diff
@@ -0,0 +1,170 @@
+import io.airlift.http.server.testing.TestingHttpServer;
+import io.trino.plugin.iceberg.IcebergQueryRunner;
+import io.trino.plugin.iceberg.TestIcebergParquetConnectorTest;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.TestingConnectorBehavior;
+import io.trino.tpch.TpchTable;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.rest.DelegatingRESTSessionCatalog;
+import org.assertj.core.util.Files;
+import org.testng.SkipException;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.Optional;
+
+import static io.trino.plugin.iceberg.IcebergFileFormat.PARQUET;
+import static io.trino.tpch.TpchTable.LINE_ITEM;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestTrinoRESTCatalogConnectorTest
+        extends TestIcebergParquetConnectorTest
+{
+    @Override
+    protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
+    {
+        switch (connectorBehavior) {
+            case SUPPORTS_RENAME_SCHEMA:
+            case SUPPORTS_CREATE_VIEW:
+            case SUPPORTS_COMMENT_ON_VIEW:
+            case SUPPORTS_CREATE_MATERIALIZED_VIEW:
+            case SUPPORTS_RENAME_MATERIALIZED_VIEW:
+                return false;
+            default:
+                return super.hasBehavior(connectorBehavior);
+        }
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        File warehouseLocation = Files.newTemporaryFolder();
+        warehouseLocation.deleteOnExit();
+
+        Catalog backend = RESTCatalogTestUtils.backendCatalog(warehouseLocation);
+
+        DelegatingRESTSessionCatalog delegatingCatalog = DelegatingRESTSessionCatalog
+                .builder().delegate(backend).build();
+
+        TestingHttpServer testServer = delegatingCatalog.testServer();
+        testServer.start();
+        closeAfterClass(testServer::stop);
+
+        return IcebergQueryRunner.builder()
+                .setBaseDataDir(Optional.of(warehouseLocation.toPath()))
+                .setIcebergProperties(
+                        ImmutableMap.<String, String>builder()
+                                .put("iceberg.file-format", "PARQUET")
+                                .put("iceberg.catalog.type", "rest")
+                                .put("iceberg.rest.uri", testServer.getBaseUrl().toString())
+                                .buildOrThrow())
+                .setInitialTables(ImmutableList.<TpchTable<?>>builder()
+                        .addAll(REQUIRED_TPCH_TABLES)
+                        .add(LINE_ITEM)
+                        .build())
+                .build();
+    }
+
+    @Override
+    @Test
+    public void testCreateSchemaWithLongName()
+    {
+        throw new SkipException("We're hitting max URI length before hitting the limit of the backing catalog");
+    }
+
+    @Override
+    @Test
+    public void testCreateTableWithLongTableName()
```

Comment:
What's the maximum name length of REST catalog? Could you override `maxTableNameLength` instead of skip?

Reply:
I don't think this makes sense to test this with the REST catalog. We are hitting 414 URI Too Long before we'd be hitting whatever the limit of the backend catalog is

Reply:
This is resolved with the latest.